### PR TITLE
Add remove functionality

### PR DESCRIPTION
**Pull request by Malcom-Charlie**
## For this project I did this and a little more:
- Removed all hardcoded items from the tasks array.
- Created new javascript file for the new functionalities 
- Implemented a function for adding a new task.
- Implemented a function for deleting a task.
- Implemented a function for editing task descriptions.
- All changes to the To Do List are be saved in local storage.

### CODE REVIEW

File: toDoList/src/index.js [hunk 1]
```diff
@@ -1,72 +1,92 @@
...
-  addTask.setAttribute('placeholder', 'Add Task');
-  addTask.classList.add('addTask');
-  todoList.appendChild(addTask);
+let tasks = [];
+
+// Function to add a new task to the array and render it
+const addTask = (taskDescription, completed = false) => {
+  tasks.push({ description: taskDescription, completed });
+  renderTasks();
+};
```
Comment (Shedrack-Sunday) [inline toDoList/src/index.js]: - [x] Based on project requirements, you were meant to create a new file for all new functionality stated. Kindly do so with all the other functions as stated in the project requirements in the snapshot below :

## Instructions:
image

File: toDoList/src/index.js [hunk 2]
```diff
@@ -1,72 +1,92 @@
...
+});
+
+// Add event listener to the removeCompletedBtn to remove all completed tasks
+removeCompletedBtn.addEventListener('click', removeCompletedTasks);
+
+// Load saved tasks from local storage
+const savedTasks = JSON.parse(localStorage.getItem('tasks'));
+if (savedTasks) {
+  tasks = savedTasks;
+  renderTasks();
```
Comment (Shedrack-Sunday) [inline toDoList/src/index.js]: - [x] As instructed on the project requirement, there should be a function for editing tasks, and this should also be included in the new file just as I pointed out in the review on line `8`. Kindly the functionality to edit 👍 

## Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @Malcom-Charlie     👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 



Highlights
✅All linters are checked
✅ Proper PR summary 
  
# Required Changes ♻️

- [x] I must admit you have really put in the effort so far in this project, however, the changes required for the new functionality were meant to be implemented ina a new file. find the instructions below :

## Instructions:
image




_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Malcom-Charlie👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- No linter errors. :heavy_check_mark:
- Highly organized code. ✔️
- GitHub flow is followed. :heavy_check_mark:



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**